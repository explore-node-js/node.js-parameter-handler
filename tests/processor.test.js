"use strict";

const Processor = require('../src/processor');
const config = require('./fixtures/package.json').extra.node_parameter_handler;
const cwd = process.cwd();

describe('processor', () => {
    describe('::resolvePath', () => {
        const processor = new Processor(config, cwd);

        [ '/', 'test', 'test/', 'test/test', '/test/test' ].forEach(path => {
            it(`should be converted '${path}' to absolute path`, () => {
                expect(processor.resolvePath(path)).toMatch(/^\/.*/);
            })
        });
    })
});
