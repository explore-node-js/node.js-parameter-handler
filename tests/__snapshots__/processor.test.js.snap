exports[`processor ::process match snapshot 1`] = `
Array [
  Object {
    "test": Object {
      "test": Object {
        "touched": undefined,
        "untouched": "untouched",
      },
      "touched": "test",
      "untouched": "untouched",
    },
    "touched": "test",
    "untouched": "untouched",
  },
  Object {
    "test": Object {
      "test": Object {
        "touched": "to be replaced",
        "untouched": "untouched",
      },
      "touched": "to be replaced",
      "untouched": "untouched",
    },
    "touched": "test",
    "untouched": "untouched",
  },
]
`;
